cpd/src/app/auth/sign-in/sign-in.component.ts

```diff
@@ -31,18 +32,29 @@ export class SignInComponent {
   });
 
   hidePassword = signal(true);
+
+  signIn: () => void;
+
   passwordInputType: InputType = InputType.password;
-  passwordMatIcon: PasswordIcon = PasswordIcon.visibility;
 
-  constructor() {
+  passwordIcon: PasswordIcon = PasswordIcon.visibility;
+
+  constructor(private authService: AuthService) {
     effect(() => {
       if(this.hidePassword()) {
         this.passwordInputType = InputType.password;
-        this.passwordMatIcon = PasswordIcon.visibility_off;
+        this.passwordIcon = PasswordIcon.visibility_off;
         return
       }
         this.passwordInputType = InputType.text;
-        this.passwordMatIcon = PasswordIcon.visibility;
+        this.passwordIcon = PasswordIcon.visibility;
+    })
+
+    this.signIn = (() => {
+        this.authService.signIn({
+        email: this.data.controls.email.value,
```

Comment:
Use destructuring assignment i stead of long property chains. Especially when data is retrieved multiple time from the same pr related source

Same everywhere